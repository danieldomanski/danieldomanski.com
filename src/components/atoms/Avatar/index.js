import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { space } from 'styled-system'

const CircledAvatar = styled.img`
  ${space};
  min-width: ${props => props.width}px;
  width: ${props => props.width}px;
  height: ${props => props.height}px;
  max-width: 100%;
  border-radius: 50%;
`

const Avatar = ({ x, y, src, mt, mb, mr, ml }) => (
  <CircledAvatar
    width={x}
    height={y}
    mt={mt}
    mb={mb}
    mr={mr}
    ml={ml}
    src={require(`../../../images/${src}`)}
    alt="avatar"
  />
)

Avatar.propTypes = {
  x: PropTypes.number,
  y: PropTypes.number,
  src: PropTypes.string.isRequired,
}

Avatar.defaultProps = {
  x: 100,
  y: 100,
}

export default Avatar
