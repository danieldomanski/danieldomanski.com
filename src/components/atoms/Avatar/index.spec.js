import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { render, cleanup } from '@testing-library/react'

import Avatar from './index'

afterEach(cleanup)

test('<Avatar> renders with different styled-system category props', () => {
  const props = {
    x: 20,
    y: 20,
    src: 'test.jpg',
  }

  const { container } = render(<Avatar {...props} />)

  expect(container.firstChild).toMatchSnapshot()
})
