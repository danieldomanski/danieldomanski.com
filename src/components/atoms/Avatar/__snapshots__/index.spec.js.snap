// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar> renders with different styled-system category props 1`] = `
.c0 {
  min-width: 20px;
  width: 20px;
  height: 20px;
  max-width: 100%;
  border-radius: 50%;
}

<img
  alt="avatar"
  class="c0"
  height="20"
  src="test-file-stub"
  width="20"
/>
`;
